File: src/api/custom-countries/src/CustomCountries.Domain/Models/Country/Country.cs
using CustomCountries.Domain.Attributes.Mongo;
using CustomCountries.Domain.Models.Mongo;
using System.Collections.Generic;

namespace CustomCountries.Domain.Models.Country
{
    [BsonCollection("countries")]
    public class Country : Document
    {
        public int CountryId { get; set; }
        public string Name { get; set; }
        public string Capital { get; set; }
        public long Area { get; set; }
        public long Population { get; set; }
        public float PopulationDensity { get; set; }
        public ICollection<TopLevelDomains> TopLevelDomains { get; set; }

        public Country()
        {
            TopLevelDomains = new HashSet<TopLevelDomains>();
        }
    }
}

File: src/webapp/countries-webapp/src/reducers/country.js
import initialState from '../constants/initialState'
import * as types from '../constants/types'

export function countries(state = initialState.countries, action) {
    switch (action.type) { 
        case types.countries.GET: {
            const { countries } = action
            let nextState = state
            for (let country of countries) {
                if (country._id && !nextState[country._id]) {
                    nextState[country._id] = country
                }
            }
            return nextState
        }
        case types.countries.UPDATE: {
            const { country } = action
            let nextState = state

            if (country && country.id && nextState[country.id]) {
                nextState[country.id].name = country.name
                nextState[country.id].capital = country.capital
                nextState[country.id].area = country.area
                nextState[country.id].population = country.population
                nextState[country.id].populationDensity = country.populationDensity
                nextState[country.id].isModified = true
            }
            return nextState
        }
        default: 
            return state
    }
}